src/main/java/org/jnode/fs/xfs/Superblock.java

```diff
@@ -92,46 +93,82 @@ public class Superblock extends XfsRecord {
      */
     public static final long XFS_SUPER_MAGIC = 0x58465342;
 
-    /**
-     * The logger.
-     */
-    private static final Logger logger = LoggerFactory.getLogger(Superblock.class);
+    // There is no need to add javadoc on the fields below,
+    // because that will make this file very big and hard to read.
+    // Please check the document for the definition.
 
-    /**
-     * The block size.
-     */
-    private final long blockSize;
+    long blockSize; // sb_blocksize
 
-    /**
-     * The inode size.
-     */
-    private final int iNodeSize;
+    long dataBlockCount; // sb_dblocks
```

Comment:
The thing that doesn't sit well with me doing this is that there is a doubling of the amount of memory used by an instance: the data buffer passed to `XfsObject` and the allocation of memory for each of these values. Ideally (possibly) it would be good if the data buffer passed in wasn't stored as a member variable and was only read into the representative member variables in this class.
That would trigger a rabbit hole of overall refactoring in the XFS code though, I think.

Reply:
I think that's OK because the superblock is pretty pretty small compared with the actual data sector, like 1k vs 1TB, so we can definitely ignore the doubling. In addition, APFS and many of other types do the same, the memory usage is tiny :)